### PR TITLE
Tech: bundle @vitejs/plugin-vue in

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: vitejs/plugin-vue has an exact dep on vite4, when we try to make it vite3 compatible. It IS actually compatible, but the version range said it is not.

## What I did

I bundled the package in, therefore the dependency disappears and can also not give a unsatisfied peerDependency warning in pnpm

### CODE REVIEW

File: code/frameworks/vue3-vite/src/preset.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ export const viteFinal: StorybookConfig['viteFinal'] = async (config, { presets
 
   // Add vue plugin if not present
   if (!(await hasVitePlugins(plugins, ['vite:vue']))) {
+    // eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (IanVS) [inline code/frameworks/vue3-vite/src/preset.ts:15]: This bugged me, so I opened https://github.com/storybookjs/storybook/pull/20440.

File: code/frameworks/vue3-vite/package.json [hunk 2]
```diff
@@ -79,6 +79,11 @@
       "./src/index.ts",
       "./src/preset.ts"
     ],
+    "externals": [
+      "util",
+      "tty",
+      "path"
```
Comment (IanVS) [inline code/frameworks/vue3-vite/package.json:85]: Why do these need to be listed as externals now?

File: scripts/prepare/bundle.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ import { exec } from '../utils/exec';
 type Formats = 'esm' | 'cjs';
 type BundlerConfig = {
   entries: string[];
-  untypedEntries: string[];
```
Comment (IanVS) [inline scripts/prepare/bundle.ts:18]: Looks like this just wasn't being used previously?

**Review by IanVS — APPROVED** [review]

File: scripts/prepare/bundle.ts (hunk 3 above)
Comment (ndelangen) [reply scripts/prepare/bundle.ts:18]: I shortly used this when doing the preview prebundling, then found a better way, just never removed this feature from the bundling code.

File: code/frameworks/vue3-vite/package.json (hunk 2 above)
Comment (ndelangen) [reply code/frameworks/vue3-vite/package.json:85]: Good question.. I do not remember 2 weeks later..
I think something relating to the package being bundled in using this, but them being node-internals?

File: code/frameworks/vue3-vite/src/preset.ts (hunk 1 above)
Comment (IanVS) [reply code/frameworks/vue3-vite/src/preset.ts:15]: This ignore line will need to be removed now that the other PR is merged.

**Pull request merged**